### DISCUSSION TITLE
v0.18.0

### DISCUSSION THREAD

**Discussion by nielsvanvelzen** [jellyfin/jellyfin-androidtv#4251] [General]
Exciting features like lyrics and media segments are now available, along with significant improvements to subtitles and more in version 0.18 of the Jellyfin for Android TV app. 

If you appreciate my work, you can show your support with a [donation trough GitHub sponsors](https://github.com/sponsors/nielsvanvelzen). Your support helps me continue improving and growing the app. Thank you!

## 🌟 Highlights

- Remove custom subtitle renderer #3825, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Implement My media \(small\) home section #4011, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Implement media segments #4052, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Rewrite all subtitle behavior #4094, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Implement lyrics #4107, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add experimental Trickplay implementation #4172, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## 🏗️ Enhancements

- Upgrade ConnectHelpAlertFragment to Jetpack Compose #3596, by [@pascalwei](https://github.com/pascalwei)
- Add option to customize seek time #3794, by [@kabel2](https://github.com/kabel2)
- Add developer preference to prefer FFmpeg for audio decoding #3818, by [@rickysixx](https://github.com/rickysixx)
- Support float preferences #3824, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Rewrite media queuing #3833, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Set seek parameter for D\-Pad to 10 seconds #3928, by [@ericleb010](https://github.com/ericleb010)
- Add resolution codec check #3953, by [@MichaelRUSF](https://github.com/MichaelRUSF)
- Avoid baking subtitles when transcoding #3980, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Increase amount of home sections from 7 to 10 #4012, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add preference for default zoom mode #4020, by [@JagerSprinkles](https://github.com/JagerSprinkles)
- Add preference to delay start of video #4025, by [@JagerSprinkles](https://github.com/JagerSprinkles)
- Add flow support for elements #4031, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Rewrite NowPlayingView in compose #4033, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Initial Jellyfin 10\.10 support #4040, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Don't skip media segments shorter than 1 second #4064, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add "ask to skip" media segment action #4068, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Use "Ask to skip" by default for intros and outros #4111, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Hide video player controls by default #4119, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Consider artist images before album artist images #4142, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Consider artist names before album artist names #4143, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Add additional information to PlaybackStartInfo #4147, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Use percentage seekbar for subtitle text size #4175, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Upgrade to Coil 3 #4190, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Unset playing item once the playback queue ends #4221, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## 💥 Crash fixes

- Fix player crashing when playback failed #4150, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Catch errors when reporting playback state #4171, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix ImageProvider always throwing #4195, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## 🔧 Bugfixes

- Fix RewriteMediaManager emitting onQueueStatusChanged with a queue when there is no audio items available #3839, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix queue not advancing #4030, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix VLC marked as watched #4032, by [@MichaelRUSF](https://github.com/MichaelRUSF)
- Fix Detekt max line length not matching Editorconfig #4058, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix skipping to next episode #4072, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix PlaybackController not always disabling default track #4098, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Automatically disable subtitles when baking fails #4112, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix adding music to existing queue #4117, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Sync custom controls with built\-in controls in leanback overlay #4118, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix video restarting immediately when baking subtitles #4122, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix some issues with lyrics #4134, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix getSubtitleDeliveryMethod in StreamInfo always returning "Encoding" #4148, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix audio track selection when baking subtitles #4154, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix exo track id parsing for format change\. #4164, by [@reacocard](https://github.com/reacocard)
- Fix position on playback stop reporting when skipping to playback end #4167, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix sorting TV shows by last played #4168, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Catch errors when loading MediaSessionService artwork #4170, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix skipping to end of video reports wrong end time #4189, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix subtitle menu not always showing "None" as first item #4196, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix skipping to end of video reports wrong end time #4197, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Use relative index matching for subtitle selection #4201, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Never load PGS externally #4202, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix player UI not hiding automatically #4203, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Really never load PGS externally #4206, by [@MichaelRUSF](https://github.com/MichaelRUSF)
- Refactor subtitle profiles to be more readable #4210, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix background of album in now playing screen when lyrics are displaying #4211, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Fix audio now playing screen always showing "1 of x" instead of the actual queue position #4217, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Specify fields in resumePlayback\(\) API request #4248, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## 🔃 Refactoring

- Fix typos #3811, by [@starsep](https://github.com/starsep)
- Use device profile builder in jellyfinPlugin #3893, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove DeviceUtils #4071, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Update package for Koin viewModel import #4082, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Rewrite usages of QueueEntry in UI #4110, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Rewrite reporting helper with SDK #4115, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- Remove runBlocking usage in HomeRowsFragment #4169, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## 💡 Everything else

- Update dependency org\.jellyfin\.sdk:jellyfin\-core to v1\.6\.3 #4228, by [@nielsvanvelzen](https://github.com/nielsvanvelzen)

## 📈 Dependency updates

- Update actions/setup\-java action by renovate\[bot\] v4\.5\.0 #4105, v4\.3\.0 #3979, v4\.2\.2 #3832
- Update Kotlin by renovate\[bot\] v2\.0\.21 #4066, v1\.9\.0 \- autoclosed #3989, v2\.0\.10 #3843
- Update gradle/actions action by renovate\[bot\] v4\.2\.1 #4227, v4\.0\.1 #3938, v4 #3857
- Update com\.android\.tools\.build:gradle by renovate\[bot\] v8\.7\.2 #4135, v8\.7\.1 #4042, v8\.6\.1 #3998, v8\.6\.0 #3948, v8\.5\.2 #3866
- Update gradle by renovate\[bot\] v8\.11\.1 #4236, v8\.11 #4194, v8\.10\.2 #4017, v8\.10\.1 \- autoclosed #3976, v8\.10 #3897
- Update github/codeql\-action action by renovate\[bot\] v3\.27\.5 #4237, v3\.27\.2 #4200, v3\.27\.1 #4184, v3\.26\.13 #4075, v3\.26\.8 #4008, v3\.26\.5 #3927, v3\.26\.4 #3920, v3\.26\.3 #3912
- Update aboutlibraries to v11\.2\.3 #3925, by renovate\[bot\]
- Update com\.android\.tools:desugar\_jdk\_libs by renovate\[bot\] v2\.1\.3 #4181, v2\.1\.2 #3960, v2\.1\.1 #3944, v2\.1\.0 #3935
- Update org\.jetbrains\.kotlinx:kotlinx\-serialization\-json by renovate\[bot\] v1\.7\.3 #4009, v1\.7\.2 #3945
- Update androidx\.fragment by renovate\[bot\] v1\.8\.5 #4123, v1\.8\.4 \- autoclosed #4043, v1\.8\.3 #3962
- Update androidx\.lifecycle by renovate\[bot\] v2\.8\.7 #4124, v2\.8\.6 #4001, v2\.8\.5 #3963
- Update androidx\.activity:activity by renovate\[bot\] v1\.9\.3 #4083, v1\.9\.2 #3964
- Update androidx\.compose\.foundation:foundation by renovate\[bot\] v1\.7\.5 #4125, v1\.7\.4 #4084, v1\.7\.3 #4044, v1\.7\.2 #4002, v1\.7\.1 #3982, v1\.7\.0 #3965
- Update androidx\.compose\.ui:ui\-tooling by renovate\[bot\] v1\.7\.5 #4126, v1\.7\.4 #4087, v1\.7\.3 #4045, v1\.7\.2 \- autoclosed #4003, v1\.7\.1 #3983, v1\.7\.0 #3966
- Update dependency io\.gitlab\.arturbosch\.detekt to v1\.23\.7 #3974, by renovate\[bot\]
- Update acra by renovate\[bot\] v5\.12\.0 #4153, v5\.11\.4 #3990
- Update koin to v4 \(major\) #3996, by renovate\[bot\]
- Update dependency androidx\.startup:startup\-runtime to v1\.2\.0 #4004, by renovate\[bot\]
- Update dependency ubuntu to v24 #4024, by renovate\[bot\]
- Update actions/upload\-artifact action to v4\.4\.3 #4061, by renovate\[bot\]
- Update dependency io\.mockk:mockk to v1\.13\.13 #4063, by renovate\[bot\]
- Update org\.jellyfin\.sdk:jellyfin\-core by renovate\[bot\] v1\.6\.2 #4222, v1\.6\.1 #4152, v1\.6\.0 #4108
- Update dependency androidx\.constraintlayout:constraintlayout to v2\.2\.0 #4127, by renovate\[bot\]
- Update dependency io\.coil\-kt\.coil3:coil\-gif to v3\.0\.3 #4215, by renovate\[bot\]
- Update appleboy/ssh\-action action to v1\.2\.0 #4223, by renovate\[bot\]
- Update coil to v3\.0\.4 #4247, by renovate\[bot\]
- chore\(deps\): update ci dependencies #3814, by renovate\[bot\]
- Update CI dependencies #3847, by renovate\[bot\]
- Update CI dependencies #3947, by renovate\[bot\]
- Update CI dependencies #3987, by renovate\[bot\]
- Update CI dependencies #4019, by renovate\[bot\]
- Update CI dependencies #4036, by renovate\[bot\]
- Update CI dependencies #4102, by renovate\[bot\]
- Update CI dependencies #4204, by renovate\[bot\]

## Contributors

- [@nielsvanvelzen](https://github.com/nielsvanvelzen)
- [@MichaelRUSF](https://github.com/MichaelRUSF)
- [@JagerSprinkles](https://github.com/JagerSprinkles)
- [@pascalwei](https://github.com/pascalwei)
- [@kabel2](https://github.com/kabel2)
- [@starsep](https://github.com/starsep)
- [@rickysixx](https://github.com/rickysixx)
- [@ericleb010](https://github.com/ericleb010)
- [@reacocard](https://github.com/reacocard)

<hr /><em>This discussion was created from the release <a href='https://github.com/jellyfin/jellyfin-androidtv/releases/tag/v0.18.0'>v0.18.0</a>.</em>

**Comment by sym0nd0** [comment]
I cannot wait to try out media segments and trick play scrubbing.

**Comment by sbackx** [comment]
Sorry for the dumb question but should the update be showing up? I have a Nvidia Shield and it doesn't think there is an update available. Google Play still shows 0.17.9

**Reply by sym0nd0** [reply]
It's not available for me yet either, I'm guessing Google have some QA process before they make updates live.

**Reply by sbackx** [reply]
I guess I could sideload since I'm very pumped about this release but I would rather get it from Google Play if I can..

Is this thread to notify everyone that it will be available soon? Or that the apk is available for testing before submitting to Google?

**Reply by sym0nd0** [reply]
This is to inform everyone of the changes. 

The version on Google Play will update but their review, to make sure it's not malware etc, and according to Googles own guidance it can take up to 7-10 days from the date of submission, I'm guessing that was Friday and that Google probably don't do a lot of reviewing over the weekend.

Hopefully be available early next week. 🤞🏻

**Reply by dredstone1** [reply]
Just say, it can sometimes take weeks, and sometimes 3 minutes.

**Comment by gigagreen880** [comment]
thx for your great work, trickplay and skipper works now perfectly and thx for the nightmode on Audio, thats really most important thing.

But can you implent a subtitle tools for the movies, like jellyfin Web or the tizen release.

![image](https://github.com/user-attachments/assets/5ed6ebdd-c001-46b1-b468-9f2dbab5f7fa)
